### ISSUE TITLE
Hiding subtasks of epics in Boards view

### ISSUE THREAD

**Issue by CRCChelsea** [ZenHubIO/support#1040]
The new (ish) features to hide and show epics and subtasks are great, but they didn't quite meet my team's needs (so close though!).

What we need is to hide all subtasks of Epics but keep all other issues and Epics visible. Basically, we use Epics as a story with sub issues that are underneath it, but when we view priorities and use the pipeline feature to pull top items into our Sprint Backlog, seeing both is redundant. We would rather hide all the subtasks of an Epic and just show the epic, but we don't want to lose all the issues that **aren't** in an Epic.

So, we want a combination of "Show all issues not in an Epic" and "Show all Epics and hide subtasks" - would that be possible?

**Comment by kendramoroz** [comment]
Hey @CRCChelsea thanks for sending this over! A great suggestion—Currently to achieve the same result, I'd recommend selecting the option **Show all issues not in an Epic** and selecting the individual Epics from the list as well. If you have quite a few epics, it's not an ideal path, but hopefully one that will work while this gets queued up in our product backlog! 

Moving over to discussion to see how others are using the Epics filter. Keep the 👍 and use cases coming!

**Comment by leemhenson** [comment]
> Currently to achieve the same result, I'd recommend selecting the option Show all issues not in an Epic and selecting the individual Epics from the list as well. If you have quite a few epics, it's not an ideal path, but hopefully one that will work while this gets queued up in our product backlog!

**Comment by leemhenson** [comment]
This has been reported previously by the looks of it:

https://github.com/ZenHubIO/support/issues/976

**Comment by kendramoroz** [comment]
Ah, sorry about that @leemhenson! I've added the original enhancement to our backlog as well as passed over the feedback on filter confusion—thanks for catching that.

**Comment by sam-github** [comment]
This is a problem for us, as well.

We want to see only the "top-level" issues: epics, and issues that are not in epics, having all the sub-tasks on the board can make it quite cluttered and hard to follow.

I have been discouraged by other team members from using the Epic feature with lots of sub-tasks (where "lots" is 6-8), because the sub-tasks are drowning out the rest of the board. I was asked to use checklists in Issues for sub-tasks as we did pre-zenhub, but the problem with them of course is that I can't track and complete a sub-task in a sprint if it exists only as a checkbox.

Basically, we are using Zenhub for its enhancements over github, like Epics, but now I'm being asked not to use them. :-(

**Comment by kendramoroz** [comment]
@sam-github Great insight here! This is on our radar as a way to improve the workflow of Epics within ZenHub. 

Today we have two different Epic filters that might be able to help clean up your view: the ability to view all Epics, no sub-tasks & the ability to show all Issues not in Epics. While you can't toggle both of these filters on to achieve the outcome you described above, you can toggle to clean up your view as you need to see both types of Issues. 

![image](https://user-images.githubusercontent.com/11036230/28328511-a5cf3718-6b9b-11e7-8a1e-7a9f36c5fc9d.png)

I'll be sure to post here as it relates to updates from the team on enhancements to Epics we have coming down the pipeline!

**Comment by sam-github** [comment]
Thanks @kendramoroz . I'm aware of those two, but we can't effectively triage the backlog by doing first the epics, then the issues not in an epic. If I could *OR* those two options, that would be truly grand.

**Comment by rashagen** [comment]
Any news on the progress of this feature. I think it's absolutely crucial filtering for anybody using epics.

**Comment by kendramoroz** [comment]
@rashagen thanks for bumping this! No immediate plans, but we're looking at various enhancements for the upcoming quarter to improve the Board experience, so I'll be sure to add an update here once we have more to share.

**Comment by rashagen** [comment]
@kendramoroz I'd say this is an absolutely crucial filtering feature. 7 👍  on the original issue above also seems to indicate I'm not alone in thinking this  :)

**Comment by anagrius** [comment]
All filtering in ZebHub is the opposite of what you need, imo. Allow me to exclude OR include. In every option, labels, epics, etc. Not AND. :( So easy to implement I just don't get it.

**Comment by leesand1** [comment]
I'm not sure if it's just me, or I'm missing something, but as a new user this is my experience:

- I have all filters removed on the board with a project containing only a single Epic
- The Epic has a number of sub-issues
- On the Epic card, there is a button on the Epic 'Show all Epic issues'
- As all issues are visible, I would expect this to default to 'Hide all Epic issues'
- Clicking this appears to has no affect

I was expecting this to toggle the visibility of the sub-tasks of that particular Epic on the board - similar to the 'Show all Epics and hide subtasks', but this isn't the case.

Please let me know if this should be moved to a separate issue!

**Comment by tdayton** [comment]
Adding to DavidAkkroo's comment of July 31: First off, my context is that I've made every issue an Epic, so that every issue can contain other issues. That's so I can use ZenHub to manage my tangled hierarchy of requirements. (Each requirement is a user story or system story or substitute your own terminology.) Here are what I think are bugs:

1. The default view of the board shows all issues--the hierarchy is expanded. What's odd and, I think, incorrect, is that all those issues display "Show all Epic issues" buttons, implying that they all are collapsed (their children are hidden, and clicking that button should show them), but obviously they are not collapsed, because all their children are shown.

2. I try to hide the children by (out of desperation) clicking the "Show all Epic issues" button of the topmost issue ("topmost" meaning in the hierarchy--the sole issue that has no parent). All the same issues still are shown, but at least now all the issues' buttons are labeled "Hide Epic issues," which matches the visibility of all the child issues. What's odd is that I clicked only the single, topmost issue's button, but all the other issues' buttons also changed.

3. Clicking the topmost issue's "Hide Epic issues" button does nothing but switch all the button labels back to "Show all Epic issues"--the same situation I started with. It does not in fact hide any issues.

4. Here is a related but different bug: Pick any lower level issue that has children--any issue that does have a parent epic. Click its "Show all Epic issues" button. That causes hiding of all issues except that one and all its descendants; all siblings and ancestors of that issue are hidden. I did not ask for that. Apparently this button's actual function is three actions: (a) Hide all issues but this one, though that is *not* what I asked for. (b) Show all immediate children of this issue, which *is* what I asked for (although per the previous bug, they already were shown). (c) Show all children of all those children, all the way to the bottom most descendant, which is *not* what I asked for. Weirdly, this button seems to be the only way to make the *board* show only the desired issue (with side effect of showing all its descendants). Instead there should be an explicitly labeled way to show only the desired issue while remaining in the board view--a control clearly labeled with that action. Whether or not to show its children while changing nothing else should be a separate control. The current control is mislabeled to appear to be only the latter ("Show all Epic issues").

5. After step 4 above, I can do something sensible: Among the descendants of that issue (the issue that has a parent that currently is hidden), is an epic that has its own child. If I click that epic's "Hide Epic issues" button, that child vanishes without changing anything else about that board view. That vanished child's parent still is visible, but so are all that parent's siblings and parent. Yay! Why does that work in this situation, but not the situations I described in the previous steps? 

6. Alas, if in the situation of step 5 I click the "Hide Epic issues" button of the topmost issue in that board view, I get a completely different result from the result in step 5. Instead I'm back to the situation in step 1--all issues are displayed, all with "Show all Epic issues" buttons.

7. WTF

**Comment by tdayton** [comment]
This is not just an enhancement issue, it is a bug.

**Comment by max-favilli** [comment]
I can't refrein myself from stressing how much needed this is.
